test/reducers/users.spec.js

```diff
@@ -0,0 +1,79 @@
+import deepFreeze from 'deep-freeze';
+import users from '../../app/assets/javascripts/reducers/users';
+import '../testHelper';
+import {
+  RECEIVE_USERS,
+  ADD_USER,
+  REMOVE_USER,
+  SORT_USERS
+} from '../../app/assets/javascripts/constants';
+
+describe('users reducer', () => {
+  it('should return initial state when no action nor state is provided', () => {
+    const newState = users(undefined, { type: null });
+    expect(newState.users).to.be.an('array');
+    expect(newState.isLoaded).to.eq(false);
+    expect(newState.sort).to.be.an('object');
+  });
+
+  it('returns the previous state and updates users array from action via RECEIVE_USERS, ADD_USER and REMOVE_USER', () => {
+    const initialState = { title: 'title', isLoaded: false };
+    deepFreeze(initialState);
+    const mockedAction = {
+      type: RECEIVE_USERS,
+      data: { course: { users: [1, 2] } }
```

Comment:
This and the following actions should have a shape that matches how the app behaves more closely. In particular, the users in this array are objects (with `id`, `username`, `role`, and other properties, if you want to get that detailed).